### PR TITLE
Fixes to #141

### PR DESCRIPTION
**Pull request by ezsh**
1. When testing NPC stats, always take NPC name from rule "OPT" property
if that is present.
2. Process "REWARD" parts of checks.

### CODE REVIEW

**Comment by JJM152** [comment]
Is this good to go now? I saw you just did another push a half hour ago?

**Comment by ezsh** [comment]
Added another fix (bullet 3 in the commit message). I ask you to check the change that always prefers NPC name from OPT, please.

File: src/000-SCRIPT_OBJ/Tasks.js [hunk 1]
```diff
@@ -143,7 +143,7 @@ App.Task = class Task {
                     }
                     break;
                 case "NPC_STAT":
-                    if (typeof NPC === 'undefined' && typeof Option !== 'undefined') NPC = Player.GetNPC(Option);
+                    if (typeof Option !== 'undefined') NPC = Player.GetNPC(Option);
```
Comment (ezsh) [inline src/000-SCRIPT_OBJ/Tasks.js]: @JJM152, please check this. No job "CHECK" in the game so far overwrites NPC name using "OPT".

Comment (JJM152) [reply src/000-SCRIPT_OBJ/Tasks.js]: It's not ringing a bell for me. I'm just going to push this through and continue testing while I work on finishing up this dice game.

Comment (ezsh) [reply src/000-SCRIPT_OBJ/Tasks.js]: OK, I continue too.

**Comment by ezsh** [comment]
It still bugged when PICK_ITEM results in money reward

**Comment by ezsh** [comment]
> It is still bugged when PICK_ITEM results in money reward

Fixed by #197.

**Pull request merged**